pkg/daemon/daemon.go

```diff
@@ -1470,9 +1470,16 @@ func (dn *Daemon) removeUpdateInProgressTaint(ctx context.Context) error {
 	})
 }
 
-func (dn *Daemon) isLayeredNode() bool {
+func (dn *Daemon) isLayeredNode() *CoreOSDaemon {
 	_, ok := dn.node.Annotations[constants.DesiredImageConfigAnnotationKey]
-	return ok
+	if ok {
+		if dn.os.IsCoreOSVariant() {
+			return &CoreOSDaemon{dn}
+		} else {
+			glog.Info("refusing to treat a non-CoreOS node as layered. You should consider using CoreOS though, it's pretty cool")
```

Comment:
also while i agree, probably fix this message so it never goes into prod ;)

Reply:
Haha okay will do

Reply:
Removed